components/raftstore/src/store/fsm/peer.rs

```diff
@@ -899,6 +899,20 @@ where
         region_local_state.set_region(self.region().clone());
         self_report.set_region_state(region_local_state);
         self_report.set_is_force_leader(self.fsm.peer.force_leader.is_some());
+        match self.fsm.peer.get_store().entries(
+            self.fsm.peer.raft_group.store().commit_index() + 1,
+            self.fsm.peer.get_store().last_index() + 1,
+            NO_LIMIT,
+            GetEntriesContext::empty(false),
+        ) {
+            Ok(entries) => {
+                for entry in entries {
+                    let ctx = ProposalContext::from_bytes(&entry.context);
+                    self_report.set_has_commit_merge(ctx.contains(ProposalContext::COMMIT_MERGE));
```

Comment:
What if the commit merge entry is in the middle?

Reply:
Sorry, I was being dumb 😢 Updated.